internal/app/go-filecoin/paymentchannel/manager_test.go

```diff
@@ -336,18 +332,23 @@ func setupViewerManager(ctx context.Context, t *testing.T, root cid.Cid) (*paych
 }
 
 func requireSetupPaymentChannel(t *testing.T, testAPI *paychtest.FakePaymentChannelAPI, m *Manager, balance abi.TokenAmount) (address.Address, address.Address, address.Address, uint64) {
-
 	clientAddr := spect.NewIDAddr(t, 901)
 	minerAddr := spect.NewIDAddr(t, 902)
 	paychUniqueAddr := spect.NewActorAddr(t, "abcd123")
 	blockHeight := uint64(1234)
 
 	testAPI.ExpectedMsgCid, testAPI.ExpectedResult = paychtest.GenCreatePaychActorMessage(t, clientAddr, minerAddr, paychUniqueAddr, balance, exitcode.Ok, blockHeight)
 
-	addr, mcid, err := m.CreatePaymentChannel(clientAddr, minerAddr, balance)
+	_, mcid, err := m.CreatePaymentChannel(clientAddr, minerAddr, balance)
 	require.NoError(t, err)
-	require.Equal(t, addr, paychUniqueAddr)
+
+	// give goroutine a chance to update channel store
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Similar questions to before, why can't we get this synchronously? Can we avoid goroutines? Or if necessary, get a latch or similar barrier to await?

Please reclassify every test calling into here as an integration test if this sleep remains.